packages/salesforce-adapter/src/change_validators/currency_iso_codes.ts

```diff
@@ -0,0 +1,119 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {
+  Change,
+  ChangeError,
+  ChangeValidator, ElemID, getChangeData,
+  InstanceElement,
+  isAdditionOrModificationChange,
+  isInstanceChange, isInstanceElement,
+} from '@salto-io/adapter-api'
+import { collections, values } from '@salto-io/lowerdash'
+import _ from 'lodash'
+import { logger } from '@salto-io/logging'
+import {
+  CURRENCY_CODE_TYPE_NAME,
+  CURRENCY_ISO_CODE,
+  FIELD_ANNOTATIONS,
+  INSTANCE_FULL_NAME_FIELD,
+  SALESFORCE,
+} from '../constants'
+
+const log = logger(module)
+
+const { awu } = collections.asynciterable
+const { isDefined } = values
+const { makeArray } = collections.array
+
+type CurrencyIsoCodesInstance = InstanceElement & {
+  value: InstanceElement['value'] & {
+    [FIELD_ANNOTATIONS.VALUE_SET]: {
+      [INSTANCE_FULL_NAME_FIELD]: string
+    }[]
+  }
+}
+
+type InstanceWithCurrencyIsoCode = InstanceElement & {
+  value: InstanceElement['value'] & {
+    [CURRENCY_ISO_CODE]: string
+  }
+}
+
+const isInstanceWithCurrencyIsoCode = (instance: InstanceElement): instance is InstanceWithCurrencyIsoCode => (
+  _.isString(instance.value[CURRENCY_ISO_CODE])
+)
+
+const isValidCurrencyIsoCodesInstance = (instance: InstanceElement): instance is CurrencyIsoCodesInstance => {
```

Comment:
Nit: I think valid is redundant in a type gaurd
```suggestion
const isCurrencyIsoCodesInstance = (instance: InstanceElement): instance is CurrencyIsoCodesInstance => {
```

Reply:
v